### PR TITLE
refactor(poe): using_person_on_events -> person_on_events_mode

### PR DESCRIPTION
**Pull request by yakkomajuri**
A few refactors here to lay the ground work for PoE v2. A top-down approach compared to #14747's bottom-up approach.

The idea here is that by passing the mode we'll avoid a second param `using_persons_on_events_v2` and having everything downstream need to have logic for v2 vs v1 etc.

This is meant to be a no-op refactor! Need **careful reviewing** though, would appreciate someone helping me look over all the instances that changed. 

A bunch of the checks for `!= disabled` and `== disabled` will soon become checks for v1 vs v2

### CODE REVIEW

File: ee/clickhouse/queries/funnels/funnel_correlation.py [hunk 1]
```diff
@@ -539,20 +539,20 @@ def _get_properties_prop_clause(self):
                 param_name = f"property_name_{index}"
                 if self._filter.aggregation_group_type_index is not None:
                     expression, _ = get_property_string_expr(
-                        "groups" if not self._team.person_on_events_querying_enabled else "events",
+                        "groups" if not self._team.person_on_events_mode != PersonOnEventsMode.DISABLED else "events",
```
Comment (yakkomajuri) [inline ee/clickhouse/queries/funnels/funnel_correlation.py:542]:
```suggestion
                        "groups" if self._team.person_on_events_mode == PersonOnEventsMode.DISABLED else "events",
```

File: posthog/test/test_team.py [hunk 2]
```diff
@@ -140,10 +141,10 @@ def test_team_on_self_hosted_uses_instance_setting_to_determine_person_on_events
...
             with override_instance_config("PERSON_ON_EVENTS_ENABLED", True):
                 team = Team.objects.create_with_data(organization=self.organization)
-                self.assertTrue(team.person_on_events_querying_enabled)
+                self.assertEqual(team.person_on_events_mode, PersonOnEventsMode.V1_ENABLED)
                 mock_feature_enabled.assert_not_called()
 
             with override_instance_config("PERSON_ON_EVENTS_ENABLED", False):
                 team = Team.objects.create_with_data(organization=self.organization)
-                self.assertFalse(team.person_on_events_querying_enabled)
+                self.assertTrue(team.person_on_events_mode, PersonOnEventsMode.DISABLED)
```
Comment (yakkomajuri) [inline posthog/test/test_team.py:149]:
```suggestion
                self.assertEqual(team.person_on_events_mode, PersonOnEventsMode.DISABLED)
```

File: posthog/queries/trends/trends_event_query_base.py [hunk 3]
```diff
@@ -30,10 +31,8 @@ def get_query_base(self) -> Tuple[str, Dict[str, Any]]:
 
         prop_query, prop_params = self._get_prop_groups(
             self._filter.property_groups.combine_property_group(PropertyOperatorType.AND, self._entity.property_groups),
-            person_properties_mode=PersonPropertiesMode.DIRECT_ON_EVENTS
-            if self._using_person_on_events
-            else PersonPropertiesMode.USING_PERSON_PROPERTIES_COLUMN,
-            person_id_joined_alias=f"{self.DISTINCT_ID_TABLE_ALIAS if not self._using_person_on_events else self.EVENT_TABLE_ALIAS}.person_id",
+            person_properties_mode=get_person_properties_mode(self._team),
+            person_id_joined_alias=f"{self.DISTINCT_ID_TABLE_ALIAS if not self._person_on_events_mode != PersonOnEventsMode.DISABLED else self.EVENT_TABLE_ALIAS}.person_id",
```
Comment (yakkomajuri) [inline posthog/queries/trends/trends_event_query_base.py:35]:
```suggestion
            person_id_joined_alias=f"{self.DISTINCT_ID_TABLE_ALIAS if self._person_on_events_mode == PersonOnEventsMode.DISABLED else self.EVENT_TABLE_ALIAS}.person_id",
```

File: posthog/queries/foss_cohort_query.py [hunk 4]
```diff
@@ -303,11 +304,11 @@ def _get_behavior_subquery(self) -> Tuple[str, Dict[str, Any], str]:
         if self._should_join_behavioral_query:
 
             _fields = [
-                f"{self.DISTINCT_ID_TABLE_ALIAS if not self._using_person_on_events else self.EVENT_TABLE_ALIAS}.person_id AS person_id"
+                f"{self.DISTINCT_ID_TABLE_ALIAS if not self._person_on_events_mode else self.EVENT_TABLE_ALIAS}.person_id AS person_id"
```
Comment (yakkomajuri) [inline posthog/queries/foss_cohort_query.py:307]:
```suggestion
                f"{self.DISTINCT_ID_TABLE_ALIAS if self._person_on_events_mode == PersonOnEventsMode.DISABLED else self.EVENT_TABLE_ALIAS}.person_id AS person_id"
```

File: posthog/queries/foss_cohort_query.py [hunk 5]
```diff
@@ -303,11 +304,11 @@ def _get_behavior_subquery(self) -> Tuple[str, Dict[str, Any], str]:
...
             ]
             _fields.extend(self._fields)
 
-            if self.should_pushdown_persons and self._using_person_on_events:
+            if self.should_pushdown_persons and self._person_on_events_mode:
```
Comment (yakkomajuri) [inline posthog/queries/foss_cohort_query.py:311]:
```suggestion
            if self.should_pushdown_persons and self._person_on_events_mode != PersonOnEventsMode.DISABLED:
```

File: posthog/queries/foss_cohort_query.py [hunk 6]
```diff
@@ -463,7 +464,7 @@ def get_performed_event_multiple(self, prop: Property, prepend: str, idx: int) -
         )
 
     def _determine_should_join_distinct_ids(self) -> None:
-        self._should_join_distinct_ids = not self._using_person_on_events
+        self._should_join_distinct_ids = not self._person_on_events_mode
```
Comment (yakkomajuri) [inline posthog/queries/foss_cohort_query.py:467]:
```suggestion
        self._should_join_distinct_ids = self._person_on_events_mode == PersonOnEventsMode.DISABLED
```

File: posthog/queries/paths/paths_event_query.py [hunk 7]
```diff
@@ -24,9 +25,7 @@ def get_query(self) -> Tuple[str, Dict[str, Any]]:
         funnel_paths_join = ""
         funnel_paths_filter = ""
 
-        person_id = (
-            f"{self.DISTINCT_ID_TABLE_ALIAS if not self._using_person_on_events else self.EVENT_TABLE_ALIAS}.person_id"
-        )
+        person_id = f"{self.DISTINCT_ID_TABLE_ALIAS if not self._person_on_events_mode != PersonOnEventsMode.DISABLED else self.EVENT_TABLE_ALIAS}.person_id"
```
Comment (yakkomajuri) [inline posthog/queries/paths/paths_event_query.py:28]:
```suggestion
        person_id = f"{self.DISTINCT_ID_TABLE_ALIAS if self._person_on_events_mode == PersonOnEventsMode.DISABLED else self.EVENT_TABLE_ALIAS}.person_id"
```

File: posthog/queries/trends/breakdown.py [hunk 8]
```diff
@@ -75,21 +75,21 @@ def __init__(
...
         self.column_optimizer = column_optimizer or ColumnOptimizer(self.filter, self.team_id)
-        self.using_person_on_events = using_person_on_events
+        self.person_on_events_mode = person_on_events_mode
 
     @cached_property
     def _person_properties_mode(self) -> PersonPropertiesMode:
         return (
             PersonPropertiesMode.USING_PERSON_PROPERTIES_COLUMN
-            if not self.using_person_on_events
+            if not self.person_on_events_mode
```
Comment (yakkomajuri) [inline posthog/queries/trends/breakdown.py:92]:
```suggestion
            if self.person_on_events_mode == PersonOnEventsMode.DISABLED
```

File: posthog/queries/trends/breakdown.py [hunk 9]
```diff
@@ -106,7 +106,7 @@ def _props_to_filter(self) -> Tuple[str, Dict]:
         )
 
         target_properties: Optional[PropertyGroup] = props_to_filter
-        if not self.using_person_on_events:
+        if not self.person_on_events_mode:
```
Comment (yakkomajuri) [inline posthog/queries/trends/breakdown.py:109]:
```suggestion
        if self.person_on_events_mode == PersonOnEventsMode.DISABLED:
```

File: posthog/queries/trends/breakdown.py [hunk 10]
```diff
@@ -115,7 +115,7 @@ def _props_to_filter(self) -> Tuple[str, Dict]:
             table_name="e",
             person_properties_mode=self._person_properties_mode,
             person_id_joined_alias=f"{self.DISTINCT_ID_TABLE_ALIAS}.person_id"
-            if not self.using_person_on_events
+            if not self.person_on_events_mode
```
Comment (yakkomajuri) [inline posthog/queries/trends/breakdown.py:118]:
```suggestion
            if self.person_on_events_mode == PersonOnEventsMode.DISABLED
```

File: posthog/queries/trends/breakdown.py [hunk 11]
```diff
@@ -153,7 +151,7 @@ def get_query(self) -> Tuple[str, Dict, Callable]:
                 action=action,
                 table_name="e",
                 person_properties_mode=self._person_properties_mode,
-                person_id_joined_alias=f"{self.DISTINCT_ID_TABLE_ALIAS if not self.using_person_on_events else 'e'}.person_id",
+                person_id_joined_alias=f"{self.DISTINCT_ID_TABLE_ALIAS if not self.person_on_events_mode else 'e'}.person_id",
```
Comment (yakkomajuri) [inline posthog/queries/trends/breakdown.py:154]:
```suggestion
                person_id_joined_alias=f"{self.DISTINCT_ID_TABLE_ALIAS if self.person_on_events_mode == PersonOnEventsMode.DISABLED else 'e'}.person_id",
```

File: posthog/queries/trends/breakdown.py [hunk 12]
```diff
@@ -174,7 +172,7 @@ def get_query(self) -> Tuple[str, Dict, Callable]:
             "actions_query": "AND {}".format(action_query) if action_query else "",
             "event_filter": "AND event = %(event)s" if not action_query else "",
             "filters": prop_filters,
-            "null_person_filter": f"AND notEmpty(e.person_id)" if self.using_person_on_events else "",
+            "null_person_filter": f"AND notEmpty(e.person_id)" if self.person_on_events_mode else "",
```
Comment (yakkomajuri) [inline posthog/queries/trends/breakdown.py:175]:
```suggestion
            "null_person_filter": f"AND notEmpty(e.person_id)" if self.person_on_events_mode == PersonOnEventsMode.DISABLED else "",
```

File: posthog/queries/paths/paths_event_query.py (hunk 7 above)
Comment (tiina303) [inline posthog/queries/paths/paths_event_query.py:28]:
```suggestion
        person_id = f"{self.DISTINCT_ID_TABLE_ALIAS if self._person_on_events_mode == PersonOnEventsMode.DISABLED else self.EVENT_TABLE_ALIAS}.person_id"
```

File: posthog/queries/session_recordings/session_recording_list.py [hunk 13]
```diff
@@ -401,7 +401,7 @@ def _data_to_return(self, results: List[Any]) -> List[Dict[str, Any]]:
         ]
 
     def run(self, *args, **kwargs) -> SessionRecordingQueryResult:
-        self._filter.hogql_context.using_person_on_events = False
+        self._filter.hogql_context.person_on_events_mode = False
```
Comment (tiina303) [inline posthog/queries/session_recordings/session_recording_list.py:404]:
```suggestion
        self._filter.hogql_context.person_on_events_mode = PersonPropertiesMode.DISABLED
```

**Review by tiina303 — COMMENTED** [review]
noticed some, there were others too similar

File: posthog/models/team/team.py [hunk 14]
```diff
@@ -186,11 +187,27 @@ def aggregate_users_by_distinct_id(self) -> bool:
...
+        
+        return PersonOnEventsMode.DISABLED
+         
+         
+    # KLUDGE: DO NOT REFERENCE IN THE BACKEND!
+    # Keeping this property for now only to be used by the frontend in certain cases
     @property
     def person_on_events_querying_enabled(self) -> bool:
-        result = self._person_on_events_querying_enabled
-        tag_queries(person_on_events_enabled=result)
```
[the remark this answers is not in the document]
Comment (yakkomajuri) [reply posthog/models/team/team.py:192]: we have the tags set somewhere else now so it's ok

**Comment by yakkomajuri** [comment]
@neilkakkar could you maybe take a look as well?

**Review by mariusandra — APPROVED** [review]
I didn't run the code, trusting tests to do that for me since there are no UX changes I can think of testing right now... but reading through everything made sense and looks good! 👍

**Pull request merged**